docs/application-development/observability.md

```diff
@@ -58,67 +58,29 @@ For more information, see the [Go sample for metrics](https://github.com/tempora
 </TabItem>
 <TabItem value="java">
 
-To emit metrics in Java, you should use the following steps:
+To emit metrics with the Java SDK, use the [`MicrometerClientStatsReporter`](https://github.com/temporalio/sdk-java/blob/55ee7894aec427d7e384c3519732bdd61119961a/src/main/java/io/temporal/common/reporter/MicrometerClientStatsReporter.java#L34) implementation for Prometheus and other backends supported by [Micrometer](https://micrometer.io/docs), and use `WorkflowServiceStubsOptions.Builder.setMetricsScope` to set the metrics scope in your Worker or Client code.
 
-- Set the scope
-- Set the scrape end point
-- Initialize the Workflow service stub
-- Add metrics scope in the Workflow service stubs options
-- Initialize the Client
-
-In addition to the `client workflow workflowservicestubs workflowservicestubsoptions` options, set the following:
-
-```java
-import com.sun.net.httpserver.HttpServer;
-import com.uber.m3.tally.RootScopeBuilder;
-import com.uber.m3.tally.Scope;
-import com.uber.m3.util.ImmutableMap;
-import io.micrometer.prometheus.PrometheusConfig;
-import io.micrometer.prometheus.PrometheusMeterRegistry;
-import io.temporal.common.reporter.MicrometerClientStatsReporter;
-```
-
-The following code example demonstrates how to emit metrics from your Workflow.
-
-```java
-// task queue to be used for this sample
-  public static final String DEFAULT_TASK_QUEUE_NAME = "metricsqueue";
-
-  public static void main(String[] args) {
-
-    PrometheusMeterRegistry registry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
-```
-
-Set your scope.
-In this example, the scope will report once a second.
-
-```java
-Scope scope =
-        new RootScopeBuilder()
-            .reporter(new MicrometerClientStatsReporter(registry))
-            .reportEvery(com.uber.m3.util.Duration.ofSeconds(1));
-```
-
-Next, set an endpoint to allow Prometheus to scrape metrics from. In this example, the endpoint is set to a http server at `8080`.
-
-```java
-HttpServer scrapeEndpoint = MetricsUtils.startPrometheusScrapeEndpoint(registry, 8080);
-```
-
-Stopping the Worker will stop the http server that exposes the scrape endpoint.
-
-```java
-    Runtime.getRuntime().addShutdownHook(new Thread(() -> scrapeEndpoint.stop(1)));
-```
-
-Add the metrics scope to Workflow service stub options.
+The following example shows how to set the `MicrometerClientStatsReporter` for Prometheus and define the metrics scope with the `WorkflowServiceStubsOptions`.
 
 ```java
- WorkflowServiceStubsOptions stubOptions =
-        WorkflowServiceStubsOptions.newBuilder().setMetricsScope(scope).build();
+//...
+   // Set up Prometheus registry and stats reported
+   PrometheusMeterRegistry registry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
```

Comment:
I think that we should at least add an explicit comment "// your actual Prometheus config here"